scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,25 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+OPENSEARCH_INSTALLED=$(dpkg -l | grep opensearch)
```

Comment:
You are checking a installation within a deb package with dpkg.

Reply:
I'm open to suggestions but this is what I found was the easiest way to check if opensearch was already installed